homeassistant/components/bluesound/media_player.py

```diff
@@ -130,6 +144,18 @@ async def async_setup_entry(
         config_entry.runtime_data.sync_status,
     )
 
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_SET_TIMER, None, "async_increase_timer"
+    )
+    platform.async_register_entity_service(
+        SERVICE_CLEAR_TIMER, None, "async_clear_timer"
+    )
+    platform.async_register_entity_service(
+        SERVICE_JOIN, {vol.Required(ATTR_MASTER): cv.entity_id}, "async_join"
+    )
+    platform.async_register_entity_service(SERVICE_UNJOIN, None, "async_unjoin")
+
     hass.data[DATA_BLUESOUND].append(bluesound_player)
```

Comment:
I don't think we have to store them here anymore right?

Reply:
There is one remaining use case. There is a custom attribute *bluesound_group*. And there will be an attribute *group_members* once I implement the standard media_player grouping. This does not require the *BluesoundPlayer* Entity, but it requires a mapping between *entity_id/player_name* and *ip:port*. I could replace the current *hass.data* with the mapping between *entity_id -> ip:port*. But there will remain the need for a central place to store some shared information. And I haven't found a better place yet.

Reply:
You can go back and forth via entity registry right?

Reply:
To be honest: I don't know.

- Can I get the any entity from the entity registry and access its properties? I would have to get the entity and call something like `entity.ip`.
- I also need the other way around. I have to get the entity by ip and port.

Reply:
I need the `sync_status`, which is already a public property on the `BluesoundPlayer` Entity.

Reply:
Yes but what do you need it for? What data do you need to send to the device?

Reply:
The `id` used for grouping is `ip:port`. This means I need the following two things:

- I have to find the name by ip and port. I need a list of all `sync_status` for all BluesoundEntities for that.
- I have to find ip and port by entity id. I need a mapping of entity_id to sync_status for that.

Reply:
No but I mean, you don't send a full sync status to the library to the device, you only need a part of it I would assume?

Reply:
I only need ip and port.

Reply:
So why don't we fetch the config entry for that?

Reply:
I could probably do that. The config_entry has the initial `sync_status`.

Can I fetch all config_entries for the Bluesound integration? I would need to do that for each player in a group to search for a matching entity_ids based on the ips and ports.

Reply:
> I cannot use the host used during the config_flow, because I would need to resolve it every time.

Why?

Reply:
Because it's it might be a hostname and I need the underlying ip address.

Reply:
I'm still using `hass.data` in this PR. The current runtime_data is missing up-to-date information from the player. There is only the initial state of all players.

I pulled part of this PR into a separate branch. Part of that is a DataUpdateCoordinator, which would provide that information. I am pretty sure that I can remove hass.data in that PR.

Reply:
Sounds good!